fastestimator/trace/io/model_saver.py

```diff
@@ -12,10 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+from typing import Union
 
+import tensorflow as tf
+import torch
+
+from fastestimator.backend.save_model import save_model
 from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
 
 
 class ModelSaver(Trace):
-    def __init__(self, model):
-        pass
+    """save model weights based on epoch frequency during training
+
+    Args:
+        model: model instance
+        save_dir: folder path to save model
+        frequency: model saving frequency in epoch(s). Defaults to 1.
+    """
+    def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], save_dir: str, frequency: int = 1):
+        super().__init__(mode="train")
+        self.model = model
+        self.save_dir = save_dir
+        self.frequency = frequency
+
+    def on_epoch_end(self, data: Data):
+        if self.save_dir and self.system.epoch_idx % self.frequency == 0:
+            model_name = "epoch_{}".format(self.system.epoch_idx)
```

Comment:
we need more info to go into the model name. what if user training GAN is saving both their generator and discriminator?

Reply:
done. now we take model name in fe.build